devdocs/troubleshooting-guide.md

```diff
@@ -0,0 +1,198 @@
+# Developer Troubleshooting Guide
+
+1. [Troubleshooting project setup](#troubleshooting-project-setup)
+    * [Setup checklist](#setup-checklist)
+    * [Common errors and solutions](#common-errors-and-solutions)
+2. [Troubleshooting test failures](#troubleshooting-test-failures)
+    * [Optimizing IDE layout for testing](#optimizing-ide-layout-for-testing)
+    * [How automated browser tests work](#how-automated-browser-tests-work)
+    * [Getting help to resolve test failures](#getting-help-to-resolve-test-failures)
+    * [Other errors and solutions](#other-errors-and-solutions)
+        * [Errors specific to Linux](#errors-specific-to-linux)
+
+This document can help you to fix the problems encountered while contributing to TEAMMATES.Failing that, you can post in the [issue tracker](https://github.com/TEAMMATES/teammates/issues/new) to ask for help.Remember to supply as much relevant information as possible when asking for help. e.g. Which operating system are you using.
+
+## Troubleshooting project setup
+
+### Setup checklist
+
+Use this checklist to verify that your project setup is correct.
+
+1. **Undo any automatic setting changes done by Eclipse**
+
+    Make sure Eclipse hasn't modified any settings/files during set up. Run git status to ensure the repo is same as the one you cloned from Google Code. If there are any changes, do an git reset --hard and restart Eclipse.
+
+2. **Check the version of the Google plugin for Eclipse**
+
+    In Eclipse, go to Help → Install new software → What's already installed
+
+    Ensure that Google Plugin version is `4.3 or 4.4` and TestNG is installed.
+
+    ![setup-checklist-1.png](images/setup-checklist-1.png)
+
+3. **Check the text encoding settings**
+
+    Window → Preferences → General → Workspace
+
+    ![setup-checklist-2.png](images/setup-checklist-2.png)
+
+4. **Check the JDK compatibility**
+
+    Project → Properties → Java Compiler
+
+    ![setup-checklist-3.png](images/setup-checklist-3.png)
+
+5. **Check the SDK version**
```

Comment:
The picture is outdated... add a disclaimer in the title: "Check the SDK version (note: we are using version 1.9.27 instead of 1.9.4)"